crates/ra_ide/src/extend_selection.rs

```diff
@@ -88,6 +104,94 @@ fn try_extend_selection(root: &SyntaxNode, range: TextRange) -> Option<TextRange
     node.parent().map(|it| it.text_range())
 }
 
+fn extend_tokens_from_range(
+    db: &RootDatabase,
+    file_id: FileId,
+    macro_call: ast::MacroCall,
+    original_range: TextRange,
+) -> Option<TextRange> {
+    // Find all non-whitespace tokens under MacroCall
+    let all_tokens: Vec<_> = macro_call
+        .syntax()
+        .descendants_with_tokens()
+        .filter_map(|n| {
+            let token = n.as_token()?;
+            if token.kind() == WHITESPACE {
+                None
+            } else {
+                Some(token.clone())
+            }
+        })
+        .sorted_by(|a, b| Ord::cmp(&a.text_range().start(), &b.text_range().start()))
+        .collect();
```

Comment:
This looks at the all tokens in the macro call. Ideally, we should look only at the `original_rage`. I think the best way to do this would be to use a `covering_node` to get a node that covers text range, then get it's  `.first_token()` and step forward with `.next_token` until we hit the orignal range. Similarly with `.last_token` and `.prev_token`

Reply:
Fixed.